apps/system/js/app_chrome.js

```diff
@@ -414,58 +417,78 @@
     });
   };
 
+  AppChrome.prototype.getSiteUrl = function() {
+    var pageUrl = this.app.config.url;
+    var startUrl = this.app.webManifest && this.app.webManifest.start_url;
+    return startUrl ? (new URL(startUrl, pageUrl)).href : pageUrl;
+  };
+
   /**
    * Pin current site in bookmarks database.
    */
   AppChrome.prototype.pinSite = function ac_pinSite() {
     this.pin();
```

Comment:
I think we don't need this anymore, as the this should be called on the 'added' event in the database.